packages/auto-approve/src/check-pr.ts

```diff
@@ -84,20 +89,82 @@ export async function checkPRAgainstConfig(
 
     // This function checks to see if the PR is a 'special' PR,
     // i.e., if its authorship qualifies it for further checks
-    const fileAndFileRule = getTargetFile(
+    let fileAndFileRule = getTargetFile(
       changedFiles,
       rulesToValidateAgainst.author,
-      languageVersioningRules
+      languageVersioningRules,
+      0
     );
 
     // If we've found the file in this ruleset, let's run the additional checks
-    if (fileAndFileRule) {
-      // TODO: make the checks conditional based on the different kinds of conditions
-      // to be checked, i.e., runDependencyUpgradeValidation, etc.
-      if (!runVersioningValidation(fileAndFileRule)) return false;
+    // We're running each file through the check so that we can make customized checks
+    // for each file. This way, we can be as specific as possible as to how the file
+    // needs to be checked.
+    while (fileAndFileRule) {
```

Comment:
To be honest, I feel that reading this block of code little bit difficult to understand.